### PR TITLE
bpf, sock: reduce xlation for externalTrafficPolicy=Local to host_id

### PR DESCRIPTION
**Pull request by borkmann**
See commit msg. /cc @brb @gandro

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/9930)

### CODE REVIEW

File: pkg/maps/lbmap/lbmap.go [hunk 1]
```diff
@@ -49,7 +49,7 @@ type LBBPFMap struct{}
 func (*LBBPFMap) UpsertService(
 	svcID uint16, svcIP net.IP, svcPort uint16,
 	backendIDs []uint16, prevBackendCount int,
-	ipv6 bool, svcType loadbalancer.SVCType) error {
```
Comment (gandro) [inline pkg/maps/lbmap/lbmap.go:52]: You will need to update https://github.com/cilium/cilium/blob/c6e876c1ac568db853af21ccde5445cef9da1a5e/pkg/maps/lbmap/lbmap_mock.go#L36-L37 as well.

**Review by gandro — CHANGES_REQUESTED** [review]
The approach looks good to me! Since I'm still not super familiar with the logic: What happens if the connection is not translated? Is the host set up to forward the packet to the remote host or will we just drop it?

**Comment by borkmann** [comment]
> The approach looks good to me! Since I'm still not super familiar with the logic: What happens if the connection is not translated? Is the host set up to forward the packet to the remote host or will we just drop it?

The request will go through untranslated, so depending on the service type if this is routable and we end up on the remote node (e.g. nodeport), then there won't be any backends on that node and therefore the skb will be dropped.

**Review by gandro — APPROVED** [review]

**Comment by borkmann** [comment]
test-me-please

**Comment by borkmann** [comment]
test-me-please

**Comment by borkmann** [comment]
test-me-please

**Comment by borkmann** [comment]
test-me-please

**Comment by borkmann** [comment]
Jenkins provisioning issue:
```[Pipeline] // node
[Pipeline] End of Pipeline
Setting status of 84ae46ccc591b34690aee2d13d8e0136a17c43f4 to FAILURE with url https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/16574/ and message: 'Build finished. '
Using context: Cilium-Ginkgo-Tests
15:24:33  Terminated
java.net.SocketTimeoutException: Read timed out
	at java.net.SocketInputStream.socketRead0(Native Method)
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116)
	at java.net.SocketInputStream.read(SocketInputStream.java:171)
	at java.net.SocketInputStream.read(SocketInputStream.java:141)```

**Comment by borkmann** [comment]
test-me-please

**Comment by borkmann** [comment]
test-me-please

**Review by brb — CHANGES_REQUESTED** [review]
I think you need to update this as well: https://github.com/cilium/cilium/blob/master/pkg/service/service.go#L537.

**Comment by borkmann** [comment]
test-me-please

**Comment by borkmann** [comment]
test-me-please

**Pull request merged**